src/vs/workbench/contrib/positronPlots/browser/positronPlotsActions.ts

```diff
@@ -84,24 +96,113 @@ export class PlotsCopyAction extends Action2 {
 		});
 	}
 
+	private getItems(editorService: IEditorService): IQuickPickItem[] {
+		const items: IQuickPickItem[] = [
+			{
+				type: 'item',
+				id: CopyPlotTarget.VIEW,
+				label: localize('positronPlots.copyPlotsView', 'Plots View'),
+				ariaLabel: localize('positronPlots.copyPlotsView', 'Plots View'),
+			}
+		];
+
+		editorService.editors.forEach(input => {
+			if (input.editorId === PositronPlotsEditorInput.EditorID) {
+				const name = input.getName();
+				const plotId = input.resource?.toString();
+				if (plotId) {
+					items.push({
+						type: 'item',
+						id: plotId,
+						label: localize('positronPlots.copyEditor', 'Editor: {0}', name),
+						ariaLabel: localize('positronPlots.copyEditor', 'Editor: {0}', name),
+					});
+				}
+			}
+		});
+
+		return items;
+	}
+
+	private copyPlotToClipboard(plotsService: IPositronPlotsService, notificationService: INotificationService, plotId: string) {
+		plotsService.copyPlotToClipboard(plotId)
+			.then(() => {
+				notificationService.info(localize('positronPlots.plotCopied', 'Plot copied to clipboard.'));
+			})
+			.catch((error) => {
+				notificationService.error(localize('positronPlotsServiceCopyToClipboardError', 'Failed to copy plot to clipboard: {0}', error.message));
+			});
+	}
+
 	/**
-	 * Runs the action and copies the plots.
+	 * Runs the action. Shows a quick pick if no target is provided. Otherwise, copies the
+	 * target plot to the clipboard.
 	 *
 	 * @param accessor The service accessor.
 	 */
-	async run(accessor: ServicesAccessor) {
+	async run(accessor: ServicesAccessor, target?: CopyPlotTarget) {
 		const plotsService = accessor.get(IPositronPlotsService);
 		const notificationService = accessor.get(INotificationService);
-		if (plotsService.selectedPlotId) {
-			plotsService.copyPlotToClipboard()
-				.then(() => {
-					notificationService.info(localize('positronPlots.plotCopied', 'Plot copied to clipboard.'));
+		const editorService = accessor.get(IEditorService);
+		const quickPick = accessor.get(IQuickInputService);
+		const contextKeyService = accessor.get(IContextKeyService);
+
+		const context = ContextKeyExpr.equals(`config.${POSITRON_EDITOR_PLOTS}`, false);
+		if (contextKeyService.contextMatchesRules(context)) {
```

Comment:
Could we go directly through the config service since IIUC we aren't using any unique functionality of contexts?

Reply:
Of course, I forgot that it's easier to use the configuration service